haskell/test/SMT/Common/Run.hs

```diff
@@ -0,0 +1,219 @@
+import Hedgehog
+
+import Lorentz hiding (now, (>>))
+import qualified Lorentz.Contracts.Spec.FA2Interface as FA2
+import Lorentz.Test.Consumer (contractConsumer)
+import Michelson.Typed.Convert (convertContract, untypeValue)
+import Michelson.Untyped.Entrypoints (unsafeBuildEpName)
+import Morley.Nettest
+import qualified Morley.Micheline as MM
+import Morley.Nettest.Pure
+
+import Ligo.BaseDAO.Contract
+import Ligo.BaseDAO.Types
+import SMT.Model.BaseDAO.Contract
+import SMT.Model.BaseDAO.Types
+import SMT.Common.Gen
+import SMT.Common.Types
+import Test.Ligo.BaseDAO.Common.StorageHelper as StorageHelper
+import Test.Ligo.BaseDAO.Common
+
+
+runBaseDaoPropertyTest :: MkPropertyTestOption -> PropertyT IO ()
+runBaseDaoPropertyTest option@MkPropertyTestOption{..} = do
+  -- PropertyTestInput (contractCalls, ms) <- forAll (runGeneratorT (genPropertyTestInput proposalMeta) initGeneratorState)
+
+  mkPropertyTestInput <- forAll (runGeneratorT (genPropertyTestInput option) $ initGeneratorState mptoMkPropose)
+  nettestTestProp $
+    (uncapsNettestEmulated $ do
+        -- Set initial level
+        guardianContract <- originateSimple "guardian" () dummyGuardianContract
+        tokenContract <- originateSimple "TokenContract" [] dummyFA2Contract
+
+        -- Used in registry dao
+        registryDaoConsumer <- originateSimple "registryDaoConsumer" [] (contractConsumer @(MText, (Maybe MText)))
+
+        let PropertyTestInput (contractCalls, ms) = mkPropertyTestInput $ PropertyTestArg
+              { ptaGuardianAddr = unTAddress guardianContract
+              , ptaGovAddr = unTAddress tokenContract
+              , ptaViewContractAddr = registryDaoConsumer
+              }
+
+        advanceLevel (ms & msLevel)
+
+        -- Modifying `FullStorage` used by Registry/TreasuryDao
+        -- Update `msContracts` to track the contract called used by haskell implementation
+        let newMs = ms
+              { msFullStorage = mptoModifyFs (ms & msFullStorage)
+              , msContracts = Map.fromList
+                  [ ((unTAddress tokenContract), SimpleFA2ContractType $ SimpleFA2Contract [] (toMutez 0))
+                  , ((unTAddress registryDaoConsumer), OtherContractType $ OtherContract [] (toMutez 0))
+                  ]
+              }
+
+        -- Originate Dao for nettest
+        let
+          originateData = UntypedOriginateData
+            { uodName = "BaseDAO"
+            , uodBalance = toMutez 0
+            , uodStorage = untypeValue $ toVal $ (newMs & msFullStorage)
+            , uodContract = convertContract baseDAOContractLigo
+            }
+
+        daoUntyped <- originateUntyped originateData
+        let dao = TAddress @Parameter daoUntyped
+
+        -- The registry/treasury dao can run out of mutez
+        when (mptoContractType == RegistryDaoContract || mptoContractType == TreasuryDaoContract) $ do
+          sendXtz daoUntyped (unsafeBuildEpName "callCustom") ([mt|receive_xtz|], lPackValueRaw ())
```

Comment:
Shouldn't we check when this happens? It seems to me that this should trigger an error

Reply:
This happens when many `Xtz_transfer` proposals are passed. These are to be expected, so I'm not sure why we should trigger an error here.

Reply:
Unless what you mean is that, we should compare the mutez of haskell model and ligo dao as well. Currently, we don't compare this yet.

Reply:
I've added dao balance comparison now.